addons/pos_self_order/static/tests/utils/tour_utils.js

```diff
@@ -71,6 +71,25 @@ export const PosSelf = {
                 isCheck: !click,
             };
         },
+        isKioskOrderline: (name, price, qty) => {
+            return {
+                content: `Verify is there an orderline with ${name} and ${price} and ${qty}`,
+                trigger: `.o_kiosk_item_card:has(.o_kiosk_product_name:contains("${name}")):has(span.fw-bolder:contains("${price}")):has(div.d-flex:contains("${qty}"))`,
+                run: () => {},
+            };
+        },
+        isNotKioskOrderline: (name) => {
+            return {
+                content: `Verify is there no orderline with ${name}`,
+                trigger: `.o_kiosk_item_card`,
+                run: () => {
+                    const product = Array.from(document.querySelectorAll('div')).find(el => el.textContent === `${name}`);
+                    if (product){
+                        throw new TourError(`${name} should not be present in the order`);
+                    }
```

Comment:
this can be replaced by simply using  `body:not()` in the trigger

Reply:
That doesn't work, I already tried